src/components/cdPipeline/CDPipeline.tsx

```diff
@@ -1095,10 +1086,56 @@ export default class CDPipeline extends Component<CDPipelineProps, CDPipelineSta
                     </label>
                 </div>
                 {this.renderNamespaceInfo(namespaceEditable)}
+                {this.renderTriggerType()}
             </>
         )
     }
 
+    renderTriggerType() {
+      return (
+          <>
+              <p className="fs-14 fw-6 cn-9 mb-8">When do you want to deploy</p>
+              <div className="flex mb-20">
+                  <div
+                      className={`flex dc__content-start pointer w-50 pt-8 pr-16 pb-8 pl-16 br-4 mr-8 bw-1${
+                          this.state.pipelineConfig.triggerType === TriggerType.Auto ? ' bcb-1 eb-2' : ' bcn-0 en-2'
+                      }`}
+                      onClick={() => this.handleTriggerTypeChange(TriggerType.Auto)}
+                  >
+                      <BotIcon className="icon-dim-20 mr-12" />
+                      <div>
+                          <div>Automatic</div>
+                          <div>Deploy everytime a new image is received</div>
+                      </div>
+                  </div>
+                  <div
+                      className={`flex dc__content-start pointer w-50 pt-8 pr-16 pb-8 pl-16 br-4 ml-8 bw-1${
+                          this.state.pipelineConfig.triggerType === TriggerType.Manual ? ' bcb-1 eb-2' : ' bcn-0 en-2'
+                      }`}
+                      onClick={() => this.handleTriggerTypeChange(TriggerType.Manual)}
```

Comment:
same here

Reply:
data attr can be used to make the function more generic :)

Reply:
Done